sql/scan.go

```diff
@@ -682,33 +683,80 @@ func (v *qnameVisitor) Visit(expr parser.Expr, pre bool) (parser.Visitor, parser
 	if !pre || v.err != nil {
 		return nil, expr
 	}
-	qname, ok := expr.(*parser.QualifiedName)
-	if !ok {
-		return v, expr
-	}
 
-	v.err = qname.NormalizeColumnName()
-	if v.err != nil {
-		return v, expr
-	}
-	if qname.IsStar() {
+	switch t := expr.(type) {
+	case *parser.QualifiedName:
+		qname := t
+
+		v.err = qname.NormalizeColumnName()
+		if v.err != nil {
+			return nil, expr
+		}
+		if qname.IsStar() {
+			v.err = fmt.Errorf("qualified name \"%s\" not found", qname)
+			return nil, expr
+		}
+
+		desc := v.getDesc(qname)
+		if desc != nil {
+			name := qname.Column()
+			for _, col := range v.visibleCols {
+				if !equalName(name, col.Name) {
+					continue
+				}
+				return v, v.getQVal(col)
+			}
+		}
+
 		v.err = fmt.Errorf("qualified name \"%s\" not found", qname)
 		return nil, expr
-	}
 
-	desc := v.getDesc(qname)
-	if desc != nil {
-		name := qname.Column()
-		for _, col := range v.visibleCols {
-			if !equalName(name, col.Name) {
-				continue
+	case *parser.FuncExpr:
+		// Special case handling for COUNT(*) and COUNT(foo.*), expanding the star
```

Comment:
thinking about this more, would it be possible to avoid this expansion entirely? it seems that we're doing it here because the downstream machinery can't deal with unexpanded identifiers, but there's really no particular need for it, and avoiding it would save us a descriptor load.